### PR TITLE
DEVEXP-651: Try out example project

### PR DESCRIPTION
**Pull request by BillFarber**

### CODE REVIEW

File: examples/demo-project/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+The first flow reads files from your local filesystem and inserts them into the "Documents" database in
+MarkLogic. The flow uses a Batch Job to make the flow asynchronous and to send the files to MarkLogic in
+batches of 5 in order to be efficient. It includes the "from-disk" collection on each of the documents.
+
+The second queries the "Documents" database for the documents written in the first flow. It does this by
+specifying the "from-disk" collection in an otherwise empty query. Once those documents have been retrieved,
+a Batch Job writes each document into the target directory.
+
+## Running the example
+In order to run this demonstration locally, you need to make a couple of minor changes to point the file
```
Comment (rjrudin) [inline examples/demo-project/README.md:18]: I think the Mule-ish way to do this is to use a properties file and then have the file operations reference a property. The user can then just update the properties file. 

This example project for the 1.x connector shows an example of how to reference a properties file - https://github.com/marklogic/marklogic-mule-connector/blob/master/examples/importDocs/batch/project-mysql-importDocs-batch-flow.xml#L15 .

File: examples/demo-project/src/main/mule/demo.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+http://www.mulesoft.org/schema/mule/file http://www.mulesoft.org/schema/mule/file/current/mule-file.xsd
+http://www.mulesoft.org/schema/mule/http http://www.mulesoft.org/schema/mule/http/current/mule-http.xsd
+http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/ee/core/current/mule-ee.xsd
+http://www.mulesoft.org/schema/mule/batch http://www.mulesoft.org/schema/mule/batch/current/mule-batch.xsd
+http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/java/current/mule-java.xsd
+http://www.mulesoft.org/schema/mule/salesforce http://www.mulesoft.org/schema/mule/salesforce/current/mule-salesforce.xsd">
+	<marklogic:config name="MarkLogic_Connection_Configuration" doc:name="MarkLogic Connection Configuration" doc:id="a921b600-937d-43f0-aaad-bd28612b7710" >
+		<marklogic:connection host="localhost" port="8000" username="admin" password="admin"/>
+	</marklogic:config>
+	<sockets:listener-config name="Sockets_Listener_config" doc:name="Sockets Listener config" doc:id="2d0580be-c590-448b-afd8-53f15b72fa0b" >
```
Comment (rjrudin) [inline examples/demo-project/src/main/mule/demo.xml:17]: Is this needed for anything? The README mentions port 8100, and this is on port 8081, so I am assuming it's not needed. And thus the sockets connector can be removed from the dependencies too.

File: examples/demo-project/pom.xml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.mule.connectors</groupId>
+			<artifactId>mule-file-connector</artifactId>
+			<version>1.5.1</version>
+			<classifier>mule-plugin</classifier>
+		</dependency>
+		<dependency>
+			<groupId>org.mule.module</groupId>
+			<artifactId>mule-java-module</artifactId>
```
Comment (rjrudin) [inline examples/demo-project/pom.xml:74]: It looks like both this and the salesforce connector can be removed, along with their namespaces in the flow file.

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just a couple things that I think can be removed and one tweak to the flow config.

File: examples/demo-project/src/main/mule/demo.xml (hunk 2 above)
Comment (BillFarber) [reply examples/demo-project/src/main/mule/demo.xml:17]: Definitely needed. I just put the wrong port in the README.

Comment (BillFarber) [reply examples/demo-project/src/main/mule/demo.xml:17]: ugh, nvm. Removed.

**Pull request merged**